.flake8

```diff
@@ -1,11 +1,21 @@
 [flake8]
 min_python_version = 3.7.0
 max-line-length = 88
-ignore = E501, E203, W503, ANN101, ANN102, SIM106
```

Comment:
Not sure if [W503](https://www.flake8rules.com/rules/W503.html) was really relevant to ignore, especially since it is not ignored in `poetry`, but if it was, happy to update the PR to keep it.

Reply:
W503 must be ignored. It is outdated and contradicts PEP8 by recommending an anti-pattern. Further, it is not compatible with [black](https://black.readthedocs.io/en/stable/guides/using_black_with_other_tools.html#id1). However, according to the linked section in black, it is disabled by default. So I suppose it is required when overwriting `ignore`, but is not required when extending it.

Reply:
Indeed, [you're right](https://github.com/PyCQA/pycodestyle/blob/2.8.0/pycodestyle.py#L90).

It looks like the opposite rule was implemented in https://github.com/PyCQA/pycodestyle/pull/502, which should be compatible with what `black` does, but it is also ignored by default.
Switching to `extend-ignore` would make us not check `W504` anymore, and possibly other error codes that are ignored by default, so https://github.com/python-poetry/poetry-core/pull/362/commits/bea338d87ebd87331d1ee827b7c1905c48e446d7 may not be a good idea in the end.

Reply:
Never thought about that in detail. On a quick glance, it seems the default ignores are handled by black anyway. I can live with both. @python-poetry/triage Any strong opinions? `ignore` vs. `extend-ignore`?

Reply:
I like the `extend-ignore` style with the explanations.  I also trust the flake8 team enough, to go with `extend-ignore` over `ignore` and trust that things ignored by default are there for a reason. Glancing over list of default ignores linked above, I think we can go with it. Configurations are not set in stone, we can always change them later if needed.

Reply:
👍 I like `extend-ignore` for the same reasons